j2v8-debugger/src/main/java/com/alexii/j2v8debugger/Debugger.kt

```diff
@@ -1,139 +1,114 @@
 
-//users of the lib can change this value
-private val scriptsDomain = "http://app/"
-private val scriptsUrlBase get() = scriptsDomain + StethoHelper.scriptsPathPrefix
-
-//move to separate mapper class if conversion logic become complicated and used in many places
-private fun scriptIdToUrl(scriptId: String?) = scriptsUrlBase + scriptId
-private fun urlToScriptId(url: String?) = url?.removePrefix(scriptsUrlBase)
-
 /**
- * V8 JS Debugger. Name of the class and methods must match names defined in Chrome Dev Tools protocol.
- *
- * [initialize] must be called before actual debugging (adding breakpoints in Chrome DevTools).
- *  Otherwise setting breakpoint, etc. makes no effect.
- *  Above is also true for the case when debugger is paused due to pause/resume implementation as thread suspension.
+ * Debugger Domain. Name of the class and methods must match names defined in Chrome Dev Tools protocol.
  */
-class Debugger(
+@Suppress("UNUSED_PARAMETER", "unused")
+internal class Debugger(
     private val scriptSourceProvider: ScriptSourceProvider
 ) : FacebookDebuggerStub() {
     var dtoMapper: ObjectMapper = ObjectMapper()
-        @VisibleForTesting set
-        @VisibleForTesting get
-
-    //xxx: consider using WeakReference
-    /** Must be called on [v8Executor]]. */
-    var v8Debugger: DebugHandler? = null
-        private set
-        @VisibleForTesting get
-
-    //xxx: consider encapsulate BreakHandler and DebugHandler into the separate class
-    private val v8ToChromeBreakHandler = V8ToChromeDevToolsBreakHandler(::connectedPeer)
 
     /**
-     * Executor where V8 scripts are being executed on. Used by [v8Debugger].
      * Needed as @ChromeDevtoolsMethod methods are called on Stetho threads, but not v8 thread.
      *
      * XXX: consider using ThreadBound from Facebook with an implementation, which uses Executor.
      */
     private var v8Executor: ExecutorService? = null
-
+    private var v8Messenger: V8Messenger? = null
     private var connectedPeer: JsonRpcPeer? = null
+    private val breakpointsAdded = mutableListOf<String>()
 
-    companion object {
-        const val TAG = "j2v8-debugger"
-    }
-
-    fun initialize(v8Debugger: DebugHandler, v8Executor: ExecutorService) {
-        this.v8Debugger = v8Debugger
+    fun initialize(v8Executor: ExecutorService, v8Messenger: V8Messenger) {
         this.v8Executor = v8Executor
-
-        v8Debugger.addBreakHandler(v8ToChromeBreakHandler)
+        this.v8Messenger = v8Messenger
     }
 
     private fun validateV8Initialized() {
-        if (v8Executor == null || v8Debugger == null) {
-            throw IllegalStateException("Unable to set breakpoint when v8 was not initialized yet")
+        if (v8Executor == null) {
+            throw IllegalStateException("Unable to call method before v8 has been initialized")
         }
     }
 
-    @ChromeDevtoolsMethod
-    override fun enable(peer: JsonRpcPeer, params: JSONObject?) {
+    private fun onDisconnect() {
```

Comment:
As far as I see, `onDisconnect()` is not used anymore - why `peer.registerDisconnectReceiver(::onDisconnect)` was removed? 

If for some reason  it's indeed not required - perhaps we should just remove it?

Btw, I don't see this warning reported in IDE because of `@Suppress("UNUSED_PARAMETER", "unused")` on the class -
 should  we make it more fine-grained and instead apply to target method/arguments in order to prevent such issues in the future?

Reply:
Good catch.  I think it should still be calling onDisconnect to clean up the connection. I must have refactored this incorrectly and inadvertently removed the call.  I'll add it back.  Also good advice on removing class level suppression to catch this where necessary.

Reply:
👍  thanks for comments and applying the changes.